src/sentry/static/sentry/app/components/stackedBarChart.jsx

```diff
@@ -309,38 +353,61 @@ const StackedBarChart = createReactClass({
     });
 
     let children = [];
-    points.forEach(point => {
+    let markerChildren = [];
+    points.forEach((point, index) => {
       while (markers.length && markers[0].x <= point.x) {
-        children.push(this.renderMarker(markers.shift()));
+        markerChildren.push(this.renderMarker(markers.shift(), index, pointWidth));
       }
 
-      children.push(this.renderChartColumn(point, maxval, pointWidth));
+      children.push(
+        this.renderChartColumn(point, maxval, pointWidth, index, totalPoints)
+      );
     });
 
     // in bizarre case where markers never got rendered, render them last
     // NOTE: should this ever happen?
     markers.forEach(marker => {
-      children.push(this.renderMarker(marker));
+      markerChildren.push(this.renderMarker(marker));
     });
 
-    return children;
-  },
+    return (
+      <React.Fragment>
+        <StyledSvg gap={this.props.gap} overflow="visible">
+          {children}
+          {markerChildren.length ? markerChildren : null}
```

Comment:
as of today, svg renders stacking context in order. they are adding a `z-index` like feature but it isn't here yet.